Source/WebInspectorUI/UserInterface/Base/Utilities.js

```diff
@@ -1866,6 +1866,24 @@ function insertObjectIntoSortedArray(object, array, comparator)
     array.splice(insertionIndexForObjectInListSortedByFunction(object, array, comparator), 0, object);
 }
 
+async function retryUntilTruthy(fn, {delay, retries} = {})
+{
+    retries ??= 30;
```

Comment:
do we ever actually need 30 retries?  or was this just a number you picked randomly?

i am slightly concerned that this may become flaky if things get slower in the future 🤔

perhaps we should have it actually be `Infinity`?  would that actually be able to cause issues?

Reply:
> do we ever actually need 30 retries? or was this just a number you picked randomly?

Yes, it was picked at random. I increased it to 100 retries * 100ms delay for a maximum of 10 seconds of retry, which should be reasonable for the default case. 🤞 

> i am slightly concerned that this may become flaky if things get slower in the future 🤔
> 
> perhaps we should have it actually be `Infinity`? would that actually be able to cause issues?

We do want this to eventually timeout and return `null` so there's meaningful a trace to debug.

Tests have their own timeout mechanism, but this method could be used in another context without an escape hatch where an infinite wait may be harder to debug.

Where `Infinity` is truly desired, it can be configured: `retryUntil(predicate, {retries: Infinity})`